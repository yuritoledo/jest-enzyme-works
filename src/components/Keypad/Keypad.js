import React from 'react'
import PropTypes from 'prop-types'
import './Keypad.css'
import Key from '../Key/Key'

const Keypad = ({ callOperator, numbers, operators, setOperator, updateDisplay }) => {
	return (
		<div className="keypad-container">
			<div className="numbers-container">
				{numbers.map(number => (
					<Key
						key={number}
						keyAction={updateDisplay}
						keyType="number-key"
						keyValue={number}
					/>
				))}
			</div>
			<div className="operators-container">
				{operators.map(operator => (
					<Key
						key={operator}
						keyAction={setOperator}
						keyType="operator-key"
						keyValue={operator}
					/>
				))}
			</div>
			<div className="submit-container">
				<Key
					keyAction={callOperator}
					keyType="submit-key"
					keyValue="="
				/>
			</div>
		</div>
	)
}

Keypad.propTypes = {
	callOperator: PropTypes.func.isRequired,
	numbers: PropTypes.array.isRequired,
	operators: PropTypes.array.isRequired,
	setOperator: PropTypes.func.isRequired,
	updateDisplay: PropTypes.func.isRequired
}

export default Keypad