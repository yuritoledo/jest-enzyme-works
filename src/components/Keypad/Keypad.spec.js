import React from 'react'
import { shallow, mount } from 'enzyme'
import Keypad from './Keypad'

const Mockz = () => <h1>Mock</h1>

describe('Keypad', () => {
	let wrapper = shallow(<Mockz />)
	beforeEach(() => {
		wrapper = shallow(
			<Keypad
				callOperator={jest.fn()}
				setOperator={jest.fn()}
				updateDisplay={jest.fn()}
				numbers={[]}
				operators={[]}
			/>
		)
	})

	test('should render 4 div', () => {
		expect(wrapper.find('div')).toHaveLength(4)
	})

	test('should render an instance of the Key component for each index of numbers, operators, and the submit Key', () => {
		const numbers = ['0', '1']
		const operators = ['+', '-']
		const submit = 1
		const keyTotal = numbers.length + operators.length + submit
		wrapper.setProps({ numbers, operators })
		expect(wrapper.find('Key')).toHaveLength(keyTotal)
	})
})

describe('mounted Keypad', () => {
	let wrapper = shallow(<Mockz />)
	beforeEach(() => {
		wrapper = mount(
			<Keypad
				callOperator={jest.fn()}
				setOperator={jest.fn()}
				updateDisplay={jest.fn()}
				numbers={[]}
				operators={[]}
			/>
		)
	})

	test('should render correctly', () => {
		expect(wrapper).toMatchSnapshot()
	})


	test('should render the value of numbers', () => {
		wrapper.setProps({ numbers: ['1', '2', '3'] })
		expect(wrapper.find('.numbers-container').text()).toEqual('123')
	})

	test('should render value of operators', () => {
		wrapper.setProps({ operators: ['+', '-', '*', '/'] })
		expect(wrapper.find('.operators-container').text()).toEqual('+-*/')
	})
})