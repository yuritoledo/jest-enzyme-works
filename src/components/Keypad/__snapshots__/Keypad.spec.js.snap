// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mounted Keypad should render correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Keypad
    callOperator={[Function]}
    numbers={Array []}
    operators={Array []}
    setOperator={[Function]}
    updateDisplay={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "callOperator": [Function],
      "numbers": Array [],
      "operators": Array [],
      "setOperator": [Function],
      "updateDisplay": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="keypad-container"
>
        <div
                class="numbers-container"
        />
        <div
                class="operators-container"
        />
        <div
                class="submit-container"
        >
                <div
                        class="key-container submit-key"
                >
                        <p
                                class="key-value"
                        >
                                =
                        </p>
                </div>
        </div>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="numbers-container"
>
            
</div>,
          <div
            className="operators-container"
>
            
</div>,
          <div
            className="submit-container"
>
            <Key
                        keyAction={[Function]}
                        keyType="submit-key"
                        keyValue="="
            />
</div>,
        ],
        "className": "keypad-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <div
            class="numbers-container"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "numbers-container",
          },
          "ref": null,
          "rendered": Array [
            Array [],
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="operators-container"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "operators-container",
          },
          "ref": null,
          "rendered": Array [
            Array [],
          ],
          "type": "div",
        },
        Object {
          "instance": <div
            class="submit-container"
>
            <div
                        class="key-container submit-key"
            >
                        <p
                                    class="key-value"
                        >
                                    =
                        </p>
            </div>
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Key
              keyAction={[Function]}
              keyType="submit-key"
              keyValue="="
/>,
            "className": "submit-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "keyAction": [Function],
                "keyType": "submit-key",
                "keyValue": "=",
              },
              "ref": null,
              "rendered": Object {
                "instance": <div
                  class="key-container submit-key"
>
                  <p
                                    class="key-value"
                  >
                                    =
                  </p>
</div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p
                    className="key-value"
>
                    =
</p>,
                  "className": "key-container submit-key",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <p
                      class="key-value"
>
                      =
</p>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "=",
                      "className": "key-value",
                    },
                    "ref": null,
                    "rendered": Array [
                      "=",
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "callOperator": [Function],
        "numbers": Array [],
        "operators": Array [],
        "setOperator": [Function],
        "updateDisplay": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="keypad-container"
>
          <div
                    class="numbers-container"
          />
          <div
                    class="operators-container"
          />
          <div
                    class="submit-container"
          >
                    <div
                              class="key-container submit-key"
                    >
                              <p
                                        class="key-value"
                              >
                                        =
                              </p>
                    </div>
          </div>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="numbers-container"
>
              
</div>,
            <div
              className="operators-container"
>
              
</div>,
            <div
              className="submit-container"
>
              <Key
                            keyAction={[Function]}
                            keyType="submit-key"
                            keyValue="="
              />
</div>,
          ],
          "className": "keypad-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <div
              class="numbers-container"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "numbers-container",
            },
            "ref": null,
            "rendered": Array [
              Array [],
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="operators-container"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [],
              "className": "operators-container",
            },
            "ref": null,
            "rendered": Array [
              Array [],
            ],
            "type": "div",
          },
          Object {
            "instance": <div
              class="submit-container"
>
              <div
                            class="key-container submit-key"
              >
                            <p
                                          class="key-value"
                            >
                                          =
                            </p>
              </div>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Key
                keyAction={[Function]}
                keyType="submit-key"
                keyValue="="
/>,
              "className": "submit-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "keyAction": [Function],
                  "keyType": "submit-key",
                  "keyValue": "=",
                },
                "ref": null,
                "rendered": Object {
                  "instance": <div
                    class="key-container submit-key"
>
                    <p
                                        class="key-value"
                    >
                                        =
                    </p>
</div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <p
                      className="key-value"
>
                      =
</p>,
                    "className": "key-container submit-key",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <p
                        class="key-value"
>
                        =
</p>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "=",
                        "className": "key-value",
                      },
                      "ref": null,
                      "rendered": Array [
                        "=",
                      ],
                      "type": "p",
                    },
                  ],
                  "type": "div",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
